### PR TITLE
Dark theme iteration #1 (90% there)

### PR DESCRIPTION
**Pull request by jonbell-lot23**
Addresses https://github.com/RecordReplay/devtools/issues/5347.

Note: this is not the entire dark theme release. It's a foundation that gets us 90% of the way there. The remaining items are pretty small, so I'd like to make sure this "invisible release" doesn't cause any issues with light theme, then test in production, then we can get to remaining 10% polish items and cleanup in a follow-on release.

Read more [on the original ticket](https://github.com/RecordReplay/devtools/issues/5347).

Here are two looms where I confirm everything looks ok. [Dark theme](https://www.loom.com/share/2f8a4367a9bf4c1cb6a277aece41d495) and [light theme](https://www.loom.com/share/d3e2cc83460a4e1fa03891e4b9b2845d).

[Here's a new ticket](https://github.com/RecordReplay/devtools/issues/5423) for the second iteration of dark theme.

### CODE REVIEW

File: src/ui/components/NetworkMonitor/HeaderGroups.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ export function HeaderGroups({
       )}
       {headerGroups.map((headerGroup: HeaderGroup<RequestSummary>) => (
         <div
-          className="flex items-center divide-x font-normal"
+          className="flex items-center divide-x divide-themeTextField font-normal border-"
```
Comment (jasonLaster) [inline src/ui/components/NetworkMonitor/HeaderGroups.tsx:38]: what are these new classes doing?

**Review by jasonLaster — COMMENTED** [review]
Very cool!

File: src/ui/components/NetworkMonitor/HeaderGroups.tsx (hunk 1 above)
Comment (jonbell-lot23) [reply src/ui/components/NetworkMonitor/HeaderGroups.tsx:38]: I learned a new thing! [divide-x](https://tailwindcss.com/docs/divide-width) controls the border width between elements and we were using them for the vertical grabby columns in our net monitor.

 So this change is changing the colour of them from very bright gray/white to being contextual based on the theme.

Comment (jonbell-lot23) [reply src/ui/components/NetworkMonitor/HeaderGroups.tsx]: And then the border- was just a typo, so I just removed it.

**Comment by jonbell-lot23** [comment]
Hey @jasonLaster @jcmorrow and @jaril -- I've been QA'ing this PR and I think it's good to go. Can I get a thumbs up so I can push it out?

To reiterate what I said above, this is not a full release of dark mode. This is an invisible release where the only goal is "don't mess up light mode and get some stuff ready behind the scenes for dark mode to come later." Thanks!

**Review by jasonLaster — APPROVED** [review]

**Pull request merged**